### DISCUSSION TITLE
How to create unit graph

### DISCUSSION THREAD

**Discussion by vskrishnan27** [soot-oss/SootUp#697] [Q&A]
in soot:
```
Body sootBody = <body parser>;
UnitGraph graph = new BriefUnitGraph(sootBody);
Iterator<Unit> unitIt = graph.iterator();
```

How can this be replicated in SootUp?

Additionally, for my static analysis, I intend to locate a sink based on a line number from the method body. Is there a method (api) available in sootup??

do we have any document for all kind of apis Like

Stmt s getPositionInfo().getStmtPosition()

**Comment by swissiety** [comment]
Hi,
we tried to make concepts more intuitive / self-explanatory than in Soot - maybe have a look into similar method names in similar named classes. Unit has become Stmt.

```
Optional<? extends SootMethod> methodOpt = view.getMethod( methodSignature );
Body body = methodOpt.get().getBody(); // There is no 'active' body anymore
StmtGraph g = body.getStmtGraph(); // There is only the StmtGraph
for ( Stmt stmt : g ){ ... // or g.nodes() if the order is not relevant
```

For more information about the API: The [Javadoc](https://soot-oss.github.io/SootUp/apidocs/) is referenced in the README.
